src/main/java/com/redhat/devtools/lsp4ij/client/features/EditorBehaviorFeature.java

```diff
@@ -120,10 +130,7 @@ public boolean isEnableSemanticTokensFileViewProvider(@NotNull PsiFile file) {
      * otherwise false
      */
     public static boolean enableStringLiteralImprovements(@NotNull PsiFile file) {
-        return LanguageServiceAccessor.getInstance(file.getProject()).hasAny(
-                file.getVirtualFile(),
-                ls -> ls.getClientFeatures().getEditorBehaviorFeature().isEnableStringLiteralImprovements(file)
-        );
+        return isEnabled(file, EditorBehaviorFeature::isEnableStringLiteralImprovements);
```

Comment:
All of these feature flag convenience accessors now use the same caching mechanism.